core/src/main/java/org/elasticsearch/index/query/InnerHitBuilder.java

```diff
@@ -722,4 +722,16 @@ public static void extractInnerHits(QueryBuilder query, Map<String, InnerHitBuil
         }
     }
 
+    static InnerHitBuilder rewrite(InnerHitBuilder original, QueryBuilder rewrittenQuery) {
+        if (original == null) {
+            return null;
+        }
+
+        InnerHitBuilder copy = new InnerHitBuilder(original);
+        copy.query = rewrittenQuery;
+        copy.parentChildType = original.parentChildType;
+        copy.nestedPath = original.nestedPath;
```

Comment:
doesn't `new InnerHitBuilder(original)` already copy the parentChildType and nestedPath?